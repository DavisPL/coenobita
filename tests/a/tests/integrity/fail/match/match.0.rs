#![allow(unused_variables)]
#![allow(unused_assignments)]

enum Foo {
    #[cnbt::integrity({root}{root} struct ({root}{root}))]
    Bar(i32),
    
    #[cnbt::integrity({root}{root} struct ({a}{a}, {b}{b}))]
    Baz(i32, i32)
}

fn main() {
    let f = Foo::Bar(5);

    match f {
        Foo::Bar(x) => {
            #[cnbt::integrity({b}{b,c})]
            let x_ = x;
        }

        Foo::Baz(y, z) => {
            #[cnbt::integrity({a}{a})]
            let y_ = y;

            #[cnbt::integrity({b}{b})]
            let z_ = z;
        }
    }
}