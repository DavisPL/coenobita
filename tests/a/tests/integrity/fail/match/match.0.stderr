error: expected {b}{b,c}, found {bin}{*}
  --> tests/integrity/fail/match/match.0.rs:18:22
   |
18 |             let x_ = x;
   |                      ^
